client/src/procedures/primitives.rs

```diff
@@ -368,15 +410,12 @@ impl GenerateSecret for BIP39Generate {
             MnemonicLanguage::Japanese => bip39::wordlist::JAPANESE,
         };
 
-        let mnemonic = bip39::wordlist::encode(entropy.as_ref(), &wordlist).unwrap();
-
-        let mut seed = Zeroizing::new(vec![0u8; 64]);
-        let passphrase: &str = self.passphrase.as_deref().unwrap_or("");
-        let seed_mut: &mut [u8; 64] = seed.as_mut_slice().try_into().unwrap();
-        bip39::mnemonic_to_seed(&mnemonic, passphrase, seed_mut);
+        let mnemonic: bip39::Mnemonic = bip39::wordlist::encode(entropy.as_ref(), &wordlist).unwrap();
+        let mut seed = bip39::Seed::null();
+        bip39::mnemonic_to_seed(&mnemonic, &self.passphrase, &mut seed);
```

Comment:
I think I still don't really like this API. Writing to a Seed doesn't really make sense, also forces us to have this `null` function. It should just return a Seed IMO.

Reply:
Agree, the API is ugly. The main reason for it was to lower the chance of seed bytes leaking in the stack memory. The right way would be to use `Box<[u8]>` internally. But other key types (such as SecretKey) do not use `Box` internally and we accept it. I'll make a PR in crypto.

Reply:
Fixed that.